### PR TITLE
Custom header component is created (json file configure).

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/config/headerConfig.json [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      "width": "100%",
+      "textAlign": "center"
+    },
+    "headerWrapperStyle": {
+      "height": "auto",
+      "backgroundColor": "rgba(244, 233, 227, 0.21)"
+    },
+    "logo": "No",
+    "logoStyle": {},
+    "logWrapperStyle": {},
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: typo

File: src/config/headerConfig.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    "HeaderContainedButton": "No",
+    "backButton":"No",
+    "backButtonRedirectTo": "",
+    "backButtonStyle": {},
+    "logoutButton": "No",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: it should be a boolean

File: src/config/headerConfig.json [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+      "fontSize": "25px",
+      "lineHeight": "30px",
+      "width": "100%",
+      "textAlign": "center"
+    },
+    "headerWrapperStyle": {
+      "height": "auto",
+      "backgroundColor": "rgba(244, 233, 227, 0.21)"
+    },
+    "logo": "No",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: it should be boolean

File: src/config/headerConfig.json [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    "logoStyle": {},
+    "logWrapperStyle": {},
+    "HeaderContainedButton": "No",
+    "backButton":"No",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: boolean

File: src/config/headerConfig.json [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+      "textAlign": "center"
+    },
+    "headerWrapperStyle": {
+      "height": "auto",
+      "backgroundColor": "rgba(244, 233, 227, 0.21)"
+    },
+    "logo": "No",
+    "logoStyle": {},
+    "logWrapperStyle": {},
+    "HeaderContainedButton": "No",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: boolean

File: src/config/headerConfig.json [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    "backButton":"No",
+    "backButtonRedirectTo": "",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: why is this field empty??

File: src/config/headerConfig.json [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    "backButtonStyle": {},
+    "logoutButton": "No",
+    "logoutButtonRedirectTo": "",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: why is this empty??

File: src/config/headerConfig.json [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+    "logoutButtonStyle": {},
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: Do we have different styles for logout button???

File: src/config/headerConfig.json [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    "buttonWrapperStyle": {}
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: Which button is this???

File: src/config/headerConfig.json [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+      "height": "auto",
+      "backgroundColor": "rgba(244, 233, 227, 0.21)"
+    },
+    "logo": "No",
+    "logoStyle": {},
+    "logWrapperStyle": {},
+    "HeaderContainedButton": "No",
+    "backButton":"No",
+    "backButtonRedirectTo": "",
+    "backButtonStyle": {},
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: Do we use different styles for back button??

File: src/config/headerConfig.json [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+      "top": "0",
+      "left": "0",
+      "background": "var(--orange-color)",
+      "padding": "15px 0",
+      "margin": "0",
+      "color": "#fff",
+      "fontSize": "25px",
+      "lineHeight": "30px",
+      "width": "100%",
+      "textAlign": "center"
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: Do we really need to override all these styles???

File: src/config/headerConfig.json [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+    "backButton":"Yes",
+    "backButtonRedirectTo": "/admin",
+    "backButtonStyle": {},
+    "logoutButton": "Yes",
+    "logoutButtonRedirectTo": "/admin",
+    "logoutButtonStyle": {},
+    "buttonWrapperStyle": {}
+  },
+  "/home": {
+    "headerName": "जैन बाल एवं युवा संस्कार शिक्षण शिविर (चतुर्थ वर्ष)",
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: rename it to `title`

Comment (pepcus-pratik) [inline src/config/headerConfig.json]: We should not write the same title again and again

File: src/components/Header.jsx [hunk 13]
```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+
+import reactLogo1 from '../assets/images/react-logo-1.png';
```
Comment (pepcus-pratik) [inline src/components/Header.jsx]: Rename the logo

File: src/config/headerConfig.json [hunk 14]
```diff
@@ -0,0 +1,240 @@
+{
+"routeType":{
```
Comment (pepcus-pratik) [inline src/config/headerConfig.json]: No need to add an extra level with routeType.

File: src/config/headerConfig.json (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Done.

File: src/config/headerConfig.json (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Done.

File: src/config/headerConfig.json (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Done.

File: src/config/headerConfig.json (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Done.

File: src/config/headerConfig.json (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Done.

File: src/config/headerConfig.json (hunk 12 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Done.

File: src/components/Header.jsx (hunk 13 above)
Comment (BhagyashreeBhoyar) [reply src/components/Header.jsx]: Done.

File: src/config/headerConfig.json (hunk 8 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: No

File: src/config/headerConfig.json (hunk 9 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: I have removed this.

File: src/config/headerConfig.json (hunk 8 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: I have removed this.

File: src/config/headerConfig.json (hunk 10 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: No, I have removed this.

File: src/config/headerConfig.json (hunk 11 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: No, I have removed unnecessarily style.

File: src/config/headerConfig.json (hunk 12 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: I have solved this by using the default title property.

File: src/config/headerConfig.json (hunk 14 above)
Comment (BhagyashreeBhoyar) [reply src/config/headerConfig.json]: Yes.

**Pull request merged**